lib/rspec/collection_matchers/have.rb

```diff
@@ -24,6 +24,17 @@ def relativities
         }
       end
 
+      if RUBY_VERSION == '1.9.2'
+        # On Ruby 1.9.2 items that don't return an array for `to_ary`
+        # can't be flattened in arrays, we need to be able to do this
+        # to produce diffs for compound matchers, so this corrects the
+        # default implementation. Note that rspec-support has code that
+        # directly checks for pattern and prevents infinite recursion.
```

Comment:
We don't define `to_ary` on this matcher, yet it is invoked, I can't actually install 1.9.2 to debug this further, but I remember us doing something similar before. Given that 1.9.2 is eventually going away, and this is a legacy gem, I'm ok with leaving this here.